include/swift/AST/ModuleDependencies.h

```diff
@@ -152,14 +153,51 @@ class SwiftTextualModuleDependenciesStorage :
       contextHash(contextHash), isFramework(isFramework) { }
 
   ModuleDependenciesStorageBase *clone() const override {
-    return new SwiftTextualModuleDependenciesStorage(*this);
+    return new SwiftInterfaceModuleDependenciesStorage(*this);
   }
 
   static bool classof(const ModuleDependenciesStorageBase *base) {
-    return base->dependencyKind == ModuleDependenciesKind::SwiftTextual;
+    return base->dependencyKind == ModuleDependenciesKind::SwiftInterface;
   }
 };
 
+/// Describes the dependencies of a Swift module
+///
+/// This class is mostly an implementation detail for \c ModuleDependencies.
+class SwiftSourceModuleDependenciesStorage :
+  public ModuleDependenciesStorageBase {
+public:
+  /// To build a PCM to be used by this Swift module, we need to append these
+  /// arguments to the generic PCM build arguments reported from the dependency
+  /// graph.
+  const std::vector<std::string> extraPCMArgs;
+
+  /// Bridging header file, if there is one.
+  Optional<std::string> bridgingHeaderFile;
+
+  /// Swift source files that are part of the Swift module, when known.
+  std::vector<std::string> sourceFiles;
+
+  /// Source files on which the bridging header depends.
+  std::vector<std::string> bridgingSourceFiles;
+
+  /// (Clang) modules on which the bridging header depends.
+  std::vector<std::string> bridgingModuleDependencies;
```

Comment:
Minor nitpick: can we refactor these data entries to a struct and share it with `SwiftTextualModuleDependenciesStorage`?